cpp/src/digitize/digitize.cu

```diff
@@ -13,34 +13,25 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include "digitize.hpp"
 #include <cudf/digitize.hpp>
+#include <cudf/detail/digitize.hpp>
 #include <cudf/column/column_factories.hpp>
 #include <cudf/column/column_device_view.cuh>
 #include <cudf/table/row_operators.cuh>
 #include <cudf/utilities/traits.hpp>
 #include <cudf/utilities/type_dispatcher.hpp>
-#include <utilities/error_utils.hpp>
+#include <cudf/utilities/error.hpp>
 #include <thrust/binary_search.h>
 
 namespace cudf {
 
 namespace {
 
-// Initialize column with sequential values starting from 0
-auto make_sequence(size_type num_rows, rmm::mr::device_memory_resource* mr, cudaStream_t stream) {
-  auto indices = cudf::make_numeric_column(data_type{INT32}, num_rows, mask_state::UNALLOCATED, stream, mr);
-  auto view = indices->mutable_view();
-  thrust::sequence(rmm::exec_policy(stream)->on(stream), view.begin<int32_t>(), view.end<int32_t>());
-  return indices;
-}
-
-// TODO this is a specialization of row_lexicographic_comparator for single columns
-template <bool has_nulls = true>
-class less_equal_wrapper {
+template <bool has_nulls>
+class less_equal_comparator {
  public:
-  less_equal_wrapper(column_device_view lhs, column_device_view rhs,
-                                        null_order null_precedence = null_order::BEFORE)
+  less_equal_comparator(column_device_view lhs, column_device_view rhs,
```

Comment:
`<=` is not an appropriate comparator `upper/lower_bound`. It's expected to be a [StrictWearkOrdering](https://www.boost.org/sgi/stl/StrictWeakOrdering.html) comparator, which `<=` is not.

Reply:
Oh, hm. I was trying to keep the functionality of the legacy implementation which used less_equal. Should I put element_less_comparator in row_operators.cuh?

Reply:
I think the old implementation was broken then.

> Should I put element_less_comparator in row_operators.cuh?

Yeah, that sounds good.

Reply:
For reference, here's the docs in Thrust that expects the comparator to be strict weak ordering: https://thrust.github.io/doc/group__vectorized__binary__search_ga30acecaad39c7d9f299f0cc515b762ce.html#ga30acecaad39c7d9f299f0cc515b762ce